tests/utils/mock_provider.py

```diff
@@ -0,0 +1,39 @@
+import os
+from typing import Any
+
+from web3 import HTTPProvider
+from web3._utils.request import make_post_request
+from web3.providers import JSONBaseProvider
+from web3.types import RPCEndpoint, RPCResponse
+
+
+class MockProvider(JSONBaseProvider):
+    def __init__(self, mock_object: dict):
+        self._mock_object = mock_object
+        super().__init__()
+
+    def make_request(self, method: RPCEndpoint, params: Any) -> RPCResponse:
+        if method in self._mock_object:
+            result = next((x for x in self._mock_object[method] if x[0] == params), None)
+            if result is not None:
+                return result[1]
+
+            result = next((x for x in self._mock_object[method] if x[0] == 'default'), None)
+            if result is not None:
+                return result[1]
+
+        infura_project_id = os.getenv('WEB3_INFURA_PROJECT_ID')
```

Comment:
I'd prefer to not stick to specific RPC provider and don't use template for RPC URL. If Infura goes down we should be able to simply change ENV variable (without changing the code) to use another RPC provider quickly (for example Alchemy).

Reply:
It's because of brownie. It is hardcoded to use infura inside. There is a way to provide another provider, but I should brake brownie a little bit to do this. Created separate issue for this
https://github.com/lidofinance/depositor-bot/issues/62